### PR TITLE
feat: Adding compile CI

### PR DESCRIPTION
**Pull request by mm-zk**
# What ❔

* Adding CI that compiles our system contracts

## Why ❔

* CI will check that the PRs that we send

### CODE REVIEW

**Review by vladbochok — DISMISSED** [review]

File: hardhat.config.ts [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
 import '@matterlabs/hardhat-zksync-solc';
+import '@typechain/hardhat'
+
 
 const systemConfig = require('./SystemConfig.json');
 
 export default {
     zksolc: {
-        version: '1.3.11',
+        version: '1.3.14',
```
Comment (StanislavBreadless) [inline hardhat.config.ts:11]: It is a change to system contracts (not only scripts)

**Review by StanislavBreadless — CHANGES_REQUESTED** [review]

File: hardhat.config.ts (hunk 1 above)
Comment (mm-zk) [reply hardhat.config.ts:11]: good point ! - changing the PR to point at DEV.

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+on:
+  pull_request:
+
+jobs:
+  build:
+    name: Build contracts
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions-rust-lang/setup-rust-toolchain@v1
```
Comment (yorik) [inline .github/workflows/ci.yaml:11]: I think there is no need in rust here.

Comment (mm-zk) [reply .github/workflows/ci.yaml:11]: there will be (coming up in the next PR.. ;-)

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+        with:
+          toolchain: nightly-2023-04-17
+      - run: yarn
+      - run: yarn install
```
Comment (benceharomi) [inline .github/workflows/ci.yaml:15]: `yarn` with no command runs `yarn install` by default, therefore line 14 and 15 does the same thing, there is no need for both. See [yarn cli documentation](https://classic.yarnpkg.com/lang/en/docs/cli/#toc-default-command) for details.

File: .github/workflows/ci.yaml [hunk 4]
```diff
@@ -0,0 +1,19 @@
+name: "Rust CI"
+on:
+  pull_request:
+
+jobs:
+  build:
+    name: Build contracts
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
```
Comment (benceharomi) [inline .github/workflows/ci.yaml:10]: `actions/checkout@v3` uses Node v16 runtime ([see](https://github.com/actions/checkout/tree/v3#whats-new)). I would specify v18:
```
      - uses: actions/setup-node@v3
        with:
          node-version: 18.18.0  
```

File: .github/SECURITY.md
[the remark this answers is not in the document]
Comment (mm-zk) [reply .github/SECURITY.md:1]: seems that #19 was closed..

[the remark this answers is not in the document]
Comment (benceharomi) [reply .github/SECURITY.md:1]: I opened #26 instead that has the same changes

**Review by benceharomi — APPROVED** [review]

**Review by StanislavBreadless — APPROVED** [review]

File: package.json [hunk 5]
```diff
@@ -40,12 +40,12 @@
...
+    "test": "mocha test/system-contract-test.test.ts",
     "build": "hardhat compile",
     "clean": "hardhat clean",
     "fmt": "prettier --config prettier.js --write contracts/*.sol contracts/**/*.sol",
     "preprocess": "rm -rf ./bootloader/build && yarn ts-node scripts/process.ts",
     "deploy-preimages": "ts-node scripts/deploy-preimages.ts",
     "compile-yul": "ts-node scripts/compile-yul.ts"
   }
-}
+}
```
Comment (vladbochok) [inline package.json:51]: Nit: newline

**Review by vladbochok — APPROVED** [review]

**Review by benceharomi — APPROVED** [review]

**Pull request merged**